KappaGame/gui/scenes/SceneGui.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Kappa.gui.interaction;
+using Microsoft.Xna.Framework.Graphics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Kappa.gui.scenes
+{
+    abstract class SceneGui : Scene
+    {
+        protected Button[] buttons = new Button[2048];
```

Comment:
This uses too much memory for each scene. This should be a dynamic-sized array, so each scene can add their own arbitrary amount of buttons to be updated/rendered.